### PR TITLE
[cxxmodules] Fix the test-periodic-build if -Dcxxmodules=On.

### PR DESCRIPTION
**Pull request by vgvassilev**
The test should reuse the prebuilt modules in the module cache. However, due to a mismatch between the module-related flags exported by FindROOT. There is a mismatch between the -D passed by cmake.

This patch make sure that the build arguments match more closely the build setup of ROOT. Thus we can reuse the already built module files.

### CODE REVIEW

**Comment by vgvassilev** [comment]
@phsft-bot build just on slc6/clang_gcc62 with flags -Dcxxmodules=On

File: etc/cmake/FindROOT.cmake [hunk 1]
```diff
@@ -70,6 +70,14 @@ execute_process(
...
 string(REGEX REPLACE "(^|[ ]*)-I[^ ]*" "" ROOT_CXX_FLAGS "${__cflags}")
 string(REGEX REPLACE "(^|[ ]*)-I[^ ]*" "" ROOT_C_FLAGS "${__cflags}")
+if (cxxmodules)
+  # Re-add the -Wno-module-import-in-extern-c which we just filtered out.
+  # We want it because it changes the module cache hash and causes modules to be
+  # rebuilt.
+  # FIXME: We should review how we do the regex.
+  set(ROOT_CXX_FLAGS "${ROOT_CXX_FLAGS} -Wno-module-import-in-extern-c")
+  set(ROOT_C_FLAGS "${ROOT_C_FLAGS} -Wno-module-import-in-extern-c")
+endif()
```
Comment (amadio) [inline etc/cmake/FindROOT.cmake]: I don't think that this file is actually used. When you call `find_package(ROOT)` the file that is really used is `ROOTConfig.cmake`, which is installed by ROOT. This file should probably be removed from ROOT altogether.

File: test/periodic/CMakeLists.txt [hunk 2]
```diff
@@ -12,7 +12,28 @@ include(${ROOT_USE_FILE})
...
+if (ROOT_cxxmodules_FOUND)
+  # G__NdbDict.cxx includes XSElementDlg.h which requires header files from
+  # module Graf. When building Graf we build Hist->MathCore->VecCore. We fail on
+  # building VecCore because it has to be built only by the build system from
+  # bottom up (because of the specific -I flags).
+  #
+  # Ideally, this test should pick up the modules from module cache and it must
+  # not rebuild anything. Unfortunately the module Graf is built in a
+  # -DUSE_ROOT_ERROR context. Adding this -DUSE_ROOT_ERROR will allow us to
+  # reuse the module cache. The real fix is to remove the -DUSE_ROOT_ERROR from
```
Comment (amadio) [inline test/periodic/CMakeLists.txt]: If there is a "real fix", shouldn't we do that instead?

Comment (vgvassilev) [reply test/periodic/CMakeLists.txt]: Yes, we should do that but this is not possible at the moment.

File: etc/cmake/FindROOT.cmake (hunk 1 above)
Comment (vgvassilev) [reply etc/cmake/FindROOT.cmake]: Yes, indeed, I have added a fix there, too.

I've found a way out to reduce the amounts of add_definitions.

**Comment by vgvassilev** [comment]
@phsft-bot build on slc6/gcc48, slc6-i686/gcc49, centos7/clang39, centos7/gcc62, centos7/gcc7, fedora28/native, ubuntu16/native, mac1013/native, windows10/vc15 with flags -Dccache=ON

File: math/mathcore/inc/Math/Error.h [hunk 3]
```diff
@@ -27,7 +24,7 @@
    simply an std::iostream in case of stan-alone builds
 */
 
-#ifndef USE_ROOT_ERROR
+#ifdef MATHCORE_STANDALONE
```
[the remark this answers is not in the document]
Comment (vgvassilev) [reply math/mathcore/inc/Math/Error.h]: IIUC, this is an old use case and if the users (if any) will have to define this macro in their configuration.

[the remark this answers is not in the document]
Comment (henryiii) [reply math/mathcore/inc/Math/Error.h]: Could you please check with users of standalone Minuit (such as GooFit, iMinuit, etc...) by tagging at least me, Hans, and Tim before breaking things?

**Pull request merged**